ocrd_models/ocrd_models/ocrd_mets.py

```diff
@@ -215,6 +333,8 @@ def find_files(self, ID=None, fileGrp=None, pageId=None, mimetype=None, url=None
                 else:
                     if not url.fullmatch(cand_url): continue
 
+            # Note: why we instantiate a class only to find out that the local_only is set afterwards
+            # Checking local_only and url before instantiation should be better?
```

Comment:
Instantiation is fast, it just does assignment to instance vars and determines whether it's a local file in `OcrdFile.__init``. I n almost all cases, `local_only` will not be set and we will yield that file.